websites/utils.py

```diff
@@ -47,3 +47,14 @@ def get_valid_base_filename(filename: str, content_type: str) -> str:
     if filename in constants.CONTENT_FILENAMES_FORBIDDEN:
         return f"{filename}-{content_type}"
     return filename
+
+
+def set_title_salutation(website_content):
+    """Set salutation in title if salutation found in metadata and not found in title"""
+    if (
+        website_content.type == constants.CONTENT_TYPE_INSTRUCTOR
+        and website_content.metadata.get("salutation", None)
+    ):
+        salutation = website_content.metadata["salutation"].strip()
+        if not website_content.title.startswith(salutation):
+            website_content.title = f"{salutation} {website_content.title}"
```

Comment:
I think this isn't quite the right approach. Because we're only checking
`website_content.title.startswith(salutation)` there can be some strange (and
probably puzzling to the user) behavior when the `title` or the `salutation` is
edited. For instance, if I do the following:

1. create a new instructor with first name `'First'`, title `'First'`, and salutation `'prof'`
2. realize I didn't capitalize `'prof'`, change salution to `'Prof'` and save
3. the title for my instructor is now `'Prof prof First'`

<img width="316" alt="Screen Shot 2022-03-10 at 1 25 31 PM" src="https://user-images.githubusercontent.com/6207644/157730037-a98cf8b2-0a59-4ed6-bc54-0f4e45e61b6c.png">

I might also not realize that the title field is what's going to be rendered,
and think that changing the `first name` field will update what's displayed in
the rendered course - that might be something to sort out separately though.

I think the way to go about this has to be one of two things:

1. either we have to go all-in on the `title` field being computed _purely_ from `salutation` and the name fields or
2. title has to be manually editable and we discard the idea of a separate salutation.

I added a note on the issue that I think `title` is a bit of an odd naming choice here, we're using it as sort of an expanded, full form of address for the instructor rather than a title, which to me means more like 'professor', 'dr', etc.

anyhow, wondering if @pdpinch  or possible @dseaton have thoughts on this.

Reply:
Thank you for looking into this @alicewriteswrongs.
As I mentioned [here](https://github.com/mitodl/ocw-studio/issues/1073#issuecomment-1064408827), if, in the case of an instructor, the title field is hidden from the user on adding/updating an instructor and programmatically computed, then wouldn't it resolve the issue which you stated? As now, it can be rendered anywhere with accurate full name of an instructor, including salutation.

Another benefit is that the course author will not have to re-write the same thing in the title field

Reply:
yeah, my thought is that we should probably just go to setting the title field to be `salutation + " " + first_name + " " + last_name` - wondering what @pdpinch thinks about this too

Reply:
Yup, making the title either hidden or uneditable while adding/updating an instructor and setting it programmatically, as you stated, will do the job!